packages/wiz/data_stream/audit/elasticsearch/ingest_pipeline/default.yml

```
@@ -0,0 +1,325 @@
+---
+description: Pipeline for processing Audit logs.
+processors:
+  - set:
+      field: ecs.version
+      tag: set_ecs_version
+      value: '8.9.0'
+  - rename:
+      field: message
+      tag: rename_message_to_event_original
+      target_field: event.original
+      ignore_missing: true
+  - json:
+      field: event.original
+      tag: json_event_original
+      target_field: json
+      ignore_failure: true
+  - date:
+      field: json.timestamp
+      tag: date_set_timestamp
+      formats:
+        - ISO8601
+      if: ctx.json?.timestamp != null && ctx.json.timestamp != ''
+      on_failure:
+        - append:
+            field: error.message
+            value: 'Processor {{{_ingest.on_failure_processor_type}}} with tag {{{_ingest.on_failure_processor_tag}}} in pipeline {{{_ingest.on_failure_pipeline}}} failed with message: {{{_ingest.on_failure_message}}}'
+  - date:
+      field: json.timestamp
+      tag: date_set_audit_timestamp
+      target_field: wiz.audit.timestamp
+      formats:
+        - ISO8601
+      if: ctx.json?.timestamp != null && ctx.json.timestamp != ''
+      on_failure:
+        - append:
+            field: error.message
+            value: 'Processor {{{_ingest.on_failure_processor_type}}} with tag {{{_ingest.on_failure_processor_tag}}} in pipeline {{{_ingest.on_failure_pipeline}}} failed with message: {{{_ingest.on_failure_message}}}'
+  - trim:
+      field: json.action
+      tag: trim_whitespace_action
+      target_field: wiz.audit.action
+      ignore_missing: true
+      if: ctx.json?.action != ''
```

```suggestion
      if: ctx.json?.action instanceof String && ctx.json.action != ''
```
Then it cannot fail and the `on_failure` can be removed.